boa_engine/src/object/property_map.rs

```diff
@@ -44,20 +45,33 @@ enum IndexedProperties {
     /// the value field and construct the data property descriptor on demand.
     ///
     /// This storage method is used by default.
-    Dense(Vec<JsValue>),
+    Dense(ThinVec<JsValue>),
 
     /// Sparse storage this storage is used as a backup if the element keys are not continuous or the property descriptors
     /// are not data descriptors with with a value field, writable field set to `true`, configurable field set to `true`, enumerable field set to `true`.
     ///
     /// This method uses more space, since we also have to store the property descriptors, not just the value.
     /// It is also slower because we need to to a hash lookup.
-    Sparse(FxHashMap<u32, PropertyDescriptor>),
+    Sparse(Box<FxHashMap<u32, PropertyDescriptor>>),
+}
+
+unsafe impl Trace for IndexedProperties {
+    custom_trace!(this, {
+        match this {
+            Self::Dense(vec) => {
+                for elem in vec.iter() {
+                    mark(elem);
+                }
```

Comment:
I think it'd be better if we implemented `Trace` for `ThinVec` instead, which avoids duplicating `Trace` definitions.

Reply:
This would require having `ThinVec` as a dependency of the GC, or adding a feature to make it optional, right?

Reply:
Yep, if we gate it behind a "thinvec" feature it should be okay.